lib/rmt/mirror.rb

```diff
@@ -120,7 +129,9 @@ def mirror_metadata(repository_dir, repository_url, temp_metadata_dir)
 
     metadata_files
   rescue StandardError => e
-    raise RMT::Mirror::Exception.new(_('Error while mirroring metadata: %{error}') % { error: e.message })
+    if n_retry == MIRROR_METADATA_RETRIES
```

Comment:
Why do we need this conditional?

Reply:
In the scenario of those errors, where retry happens, without the checking, the error message becomes accumulative or additions, instead of the proper error message.

Reply:
I don't understand... 
So in the case when a metadata download failed 5 times, it raises a ''Failed to get repository metadata...` error. That one gets catched here an thrown up as `RMT::Mirror::Exception`. 
In that case `n_retry` is 1. So that if block would not be true, and the rescue block does nothing?

Reply:
so in that case, if no if condition in place, it would end up with
`Error while mirroring metadata: Error while mirroring metadata:....` 5 times and the error message `Mirroring metadata failed with...` at the end of the string, which is not what we want

Reply:
I think it would help to have a test where the mirroring fails 5 times, and we see that the overall mirroring is marked as failed. I'm concerned because with this `if` we're not just skipping an error message, but a `raise`.

Reply:
> I think it would help to have a test where the mirroring fails 5 times, and we see that the overall mirroring is marked as failed. 

The test has been introduced in this PR, instead of 5 times it "happens" twice

> I'm concerned because with this `if` we're not just skipping an error message, but a `raise`.

but we are not, if the exception is not raised by that error, then it would be raised: if the exception was raised by that error, then it would be raised once (not every time)

Reply:
Ok, I've tried it out by faking the exception in the code, now I understand what you mean, that should work.